common/v2/features/PurchaseMembership/helpers.ts

```diff
@@ -1,17 +1,47 @@
-import { IHexStrTransaction, ITxObject } from 'v2/types';
-import { inputValueToHex, inputGasPriceToHex, inputNonceToHex } from 'v2/services/EthService';
-import { DEFAULT_NETWORK_CHAINID } from 'v2/config';
+import { ethers } from 'ethers';
 
-import { ISimpleTxForm } from './types';
+import { ITxObject } from 'v2/types';
+import { inputValueToHex, inputGasPriceToHex, toWei } from 'v2/services/EthService';
+import { DEFAULT_NETWORK_CHAINID, DEFAULT_ASSET_DECIMAL } from 'v2/config';
+import { UnlockToken, ERC20 } from 'v2/services/EthService/contracts';
+
+import { MembershipSimpleTxFormFull } from './types';
+import { isERC20Tx } from '../SendAssets';
+
+export const createApproveTx = (payload: MembershipSimpleTxFormFull): Partial<ITxObject> => {
+  const data = ERC20.approve.encodeInput({
+    _spender: payload.membershipSelected.contractAddress,
+    _value: toWei(payload.membershipSelected.price, DEFAULT_ASSET_DECIMAL)
+  });
+
+  return {
+    // @ts-ignore Contract Address should be set if asset is ERC20
+    to: payload.asset.contractAddress,
+    from: payload.account.address,
+    data,
+    chainId: DEFAULT_NETWORK_CHAINID,
+    gasPrice: inputGasPriceToHex(payload.gasPrice),
+    value: inputValueToHex('0')
+  };
+};
+
+export const createPurchaseTx = (payload: MembershipSimpleTxFormFull): Partial<ITxObject> => {
+  const membershipSelected = payload.membershipSelected;
+
+  const weiPrice = toWei(membershipSelected.price, DEFAULT_ASSET_DECIMAL);
+  const data = UnlockToken.purchase.encodeInput({
+    _value: weiPrice,
+    _recipient: payload.account.address,
+    _referrer: ethers.constants.AddressZero,
+    _data: []
+  });
 
-export const createSimpleTxObject = (formData: ISimpleTxForm): IHexStrTransaction | ITxObject => {
   return {
-    to: formData.address,
-    value: inputValueToHex(formData.amount),
-    data: '0x',
-    gasLimit: formData.gasLimit,
```

Comment:
Do we not need this param?

Reply:
Still WIP - the MTX hook actually populates `gasLimit` but we might need to do something else.